google/cloud/spanner/internal/session_pool.cc

```diff
@@ -97,6 +97,13 @@ SessionPool::~SessionPool() {
   // `weak_ptr` to `this` they hold. Any in-progress or subsequent `lock()`
   // will now return `nullptr`, in which case no work is done.
   current_timer_.cancel();
+
+  // Send fire-and-forget `AsyncDeleteSession()` calls for all sessions.
+  for (auto const& session : sessions_) {
+    if (session->is_bad()) continue;
+    AsyncDeleteSession(cq_, session->channel()->stub, session->session_name())
+        .then([](auto result) { auto status = result.get(); });
```

Comment:
Is the `.then()` really needed?

Reply:
No, I don't believe so, but I thought it made things a little clearer, particularly if we later to chose to do any logging/tracing.